worker/buildbot_worker/commands/transfer.py

```diff
@@ -107,7 +107,8 @@ def start(self):
         self._reactor.callLater(0, self._loop, d)
 
         def _close_ok(res):
-            self.fp = None
+            if self.fp:
+                self.fp.close()
```

Comment:
I don't think it is necessary to set `self.fp` to `None`. Probably we should change the checks to see if it is `True` instead of checking for `not None`. I don't know if there is a way we could change syntax to the safer `with open(..) as self.fp` form. (Open for every write loop?) I guess this is not as easy  without changing a lot more.